Automattic/wp-calypso
# Calypso Components: Pass refs to clipboard button

### Proposed Changes

* Convert clipboard button component to typescript
* Have the clipboard component accept refs

### Testing Instructions



* Ensure that the various instances of clipboard button continue to function as expected ( there are a lot of instances, and since this is a small relatively small change, I don't think it's necessary to smoke test every single example ) I've found three instances below

#### SFTP Hosting Configuration 
<img width="755" alt="SFTP" src="https://user-images.githubusercontent.com/5414230/196532639-8f02e317-e7d5-46ed-af33-af0bb3d0117e.png">

  - Admin sidebar Settings > Hosting Configuration
  - Look at the SFTP/SSH credentials and click on the create credentials button if displayed
  - Click on the various copy buttons

#### Connect Domain Step
<img width="986" alt="ConnectDomain" src="https://user-images.githubusercontent.com/5414230/196534797-f5f2195e-5443-48df-ab4a-d82999658961.png">

  - Admin sidebar Upgrades > Domains
  - Click on "+ Add a domain"
  - Click on "Use a domain I own"
  - Enter a fake domain name into the input like `asdfjkl.com`
  - Click on "Start setup"
  - Click on "I found the domain's settings page"
  - Click on the various copy buttons to confirm behavior
 
#### Invite People Settings
<img width="1496" alt="Invite People" src="https://user-images.githubusercontent.com/5414230/196545833-d29cb71b-b6e1-4b9c-9fac-823a8d85ead2.png">

  - Navigate to http://calypso.localhost:3000/people/new/YOUR_SITE_SLUG_HERE
  - I had trouble getting the feature to display through the UI, so I actually removed the conditional in the code that was hiding the invite links
    - Open `client/my-sites/people/invite-people/index.jsx`
    - Delete lines 758 and 763
    - Return to the browser and Click on "Generate new link" button
    - Click on the copy link button
#### Pre-merge Checklist






Related to https://github.com/Automattic/wp-calypso/issues/68967
